elixir/config/runtime.exs

```diff
@@ -8,15 +8,17 @@ if config_env() == :prod do
   ###############################
 
   config :domain, Domain.Repo,
-    database: compile_config!(:database_name),
-    username: compile_config!(:database_user),
-    hostname: compile_config!(:database_host),
-    port: compile_config!(:database_port),
-    password: compile_config!(:database_password),
-    pool_size: compile_config!(:database_pool_size),
-    ssl: compile_config!(:database_ssl_enabled),
-    ssl_opts: compile_config!(:database_ssl_opts),
-    parameters: compile_config!(:database_parameters)
+  [
+    {:database, compile_config!(:database_name)},
+    {:username, compile_config!(:database_user)},
+    {:port, compile_config!(:database_port)},
+    {:pool_size, compile_config!(:database_pool_size)},
+    {:ssl, compile_config!(:database_ssl_enabled)},
+    {:ssl_opts, compile_config!(:database_ssl_opts)},
+    {:parameters, compile_config!(:database_parameters)}
+  ]
+  ++ (if System.get_env("DATABASE_PASSWORD"), do: [{:password, compile_config!(:database_password)}], else: [])
```

Comment:
Hm, I think had moved away from using `System.get_env/3` directly in configuration because it creates a difficult to reason about mess.

What do you think of instead, adding a `compile_config/3` function similar to [this one](https://github.com/firezone/firezone/blob/main/elixir/apps/domain/lib/domain/config.ex#L35) but returns `nil` instead for missing keys? Then your conditional here would use the result of that, and not the System ENV directly.

Based on that function's `@doc` comment it appears that the original author planned to write one anyhow.

Reply:
I've added this now, could you check whether my implementation is what you had intended?